src/test/scala/karazin/scala/users/group/week2/homework/HomeworkSpecification.scala

```diff
@@ -50,28 +53,30 @@ object HomeworkSpecification extends Properties("Homework"):
   }
 
   property("negation") = forAll { (rational: Rational) =>
-    ???
+    -rational == Rational(-rational.numer, rational.denom)
   }
 
   property("addition") = forAll { (left: Rational, right: Rational) =>
-    ???
+    abs((left + right).toDouble - (left.toDouble + right.toDouble)) <= SMALL_NUMBER

```

Comment:
This is a dirty test.
Why don't you compare the result numer with the sum of numers?
Compare the result numer with the sum of left and right numers. The same for denoms.
Fix it for all tests.